### DISCUSSION TITLE
Nintendo DSi port

### DISCUSSION THREAD

**Discussion by fra9000** [diasurgical/DevilutionX#6425] [Ideas]
Checking the minimum reqs of the original Diablo 1 and the Nintendo DSi specs it seems doable.
The Nintendo DSi with 133Mhz ARM9 and 16MB of RAM is much more capable compared to the original Nintendo DS/DS lite.
Nowadays with Twilight Menu there is lots of modding and attention to the DSi console with lots of ports being made.

Is there any chance by lowering resolution and color depth to give this port a try?
Also, (and this is valid for 3DS port as well), moving the UI Bar to the lower touch screen will give more space and clarity to the gameplay on the upper main screen.

Just some crazy ideas!
Thanks a lot

Francesco

PS. you are doing a very nice project here devs!

**Comment by AJenbo** [comment]
The original running on Windows 95 needed 31MB of virtual memory, where about 8-16MB was backed by RAM.

Color depth is already 8bit, you can't really get it any lower with out having to compact things as it's 1 byte pr pixel. The screen buffer only accounts for 300KB so lowering the resolution isn't going to help much either.

DSi does not provide any additional memory past the 16MB RAM so won't be able to run the game with full quality assets. You would need to make additional cuts like the PS1 edition, though maybe not as drastic. It's possible that this could be done during loading, but that would of cause significantly increase load time, alternatively a tool could be made to do the conversion offline on a computer first.

In short a DSi port is not trivial and would require someone willing to work specifically on a port for that platform for an extended time.
